lib/X25519KeyPair.js

```diff
@@ -0,0 +1,318 @@
+   * @param {string} options.controller
+   * @param {string} options.id
+   * @param {string} options.publicKeyBase58 - Base58 encoded Public Key
+   * unencoded is 32-bytes.
+   * @param {string} options.privateKeyBase58 - Base58 Private Key
+   * unencoded is 32-bytes.
+   */
+  constructor(options = {}) {
+    super(options);
+    this.type = 'X25519KeyAgreementKey2019';
+    this.privateKeyBase58 = options.privateKeyBase58;
+    this.publicKeyBase58 = options.publicKeyBase58;
+  }
+  /**
+   * Returns the Base58 encoded public key.
+   * @implements {LDKeyPair#publicKey}
+   * @readonly
+   *
+   * @returns {string} The Base58 encoded public key.
+   * @see [publicKey]{@link ./LDKeyPair.md#publicKey}
+   */
+  get publicKey() {
+    return this.publicKeyBase58;
+  }
+  /**
+   * Returns the Base58 encoded private key.
+   * @implements {LDKeyPair#privateKey}
+   * @readonly
+   *
+   * @returns {string} The Base58 encoded private key.
+   * @see [privateKey]{@link ./LDKeyPair.md#privateKey}
+   */
+  get privateKey() {
+    return this.privateKeyBase58;
+  }
+
+  /**
+   * Generates a new public/private X25519 Key Pair.
+   * Note: This is async only to match the async signature of other LDKeyPair
+   * subclasses.
+   * @example
+   * > const keyPair = await X25519KeyPair.generate();
+   * > keyPair
+   * X25519KeyPair { ...
+   * @param {KeyPairOptions} [options={}]
+   * @param {string} [options.controller]
+   * @param {string} [options.id]
+   *
+   * @returns {Promise<X25519KeyPair>} Generates a key pair.
+   */
+  static async generate(options = {}) {
+    // Each is a Uint8Array with 32-byte key
+    const {publicKey, secretKey: privateKey} = nacl.box.keyPair()
+
+    return new X25519KeyPair({
+      publicKeyBase58: base58.encode(publicKey),
+      privateKeyBase58: base58.encode(privateKey),
+      ...options
+    });
+  }
+
+  /**
+   * Creates an X25519KeyPair Key Pair from an existing key (constructor method)
+   * @example
+   * > const options = {
+   *   id,
+   *   controller,
+   *   publicKeyBase58,
+   *   privateKeyBase58: privateKey
+   * };
+   * > const key = await X25519KeyPair.from(options);
+   * > key
+   * X25519KeyPair { ...
+   * @param {Object} options
+   * @param {string} [options.privateKeyBase58] - A Base58 encoded Private key
+   *
+   * @returns {X25519KeyPair} An X25519 Key Pair.
+   */
+  static async from(options) {
+    return new X25519KeyPair(options);
+  }
+
+  static fromEdKeyPair(edKeyPair) {
+    const xKey = new X25519KeyPair({
+      controller: edKeyPair.controller
+    });
+
+    xKey.publicKeyBase58 = X25519KeyPair
+      .convertFromEdPublicKey(edKeyPair.publicKeyBase58);
+
+    if(edKeyPair.privateKeyBase58) {
+      xKey.privateKeyBase58 = X25519KeyPair
+        .convertFromEdPrivateKey(edKeyPair.privateKeyBase58)
+    }
+
+    return xKey;
+  }
+
+  /**
+   * @param {string} edPublicKeyBase58 - base58 encoded Ed25519 Public key
+   * @returns {string} base58 encoded X25519 Public key
```

Comment:
Linting stuff again; inconsistent use of `-`, no periods, etc.

Reply:
Fixed.